nomad/structs/network.go

```diff
@@ -167,38 +227,42 @@ func (idx *NetworkIndex) AddAllocs(allocs []*Allocation) (collide bool) {
 			// Only look at AllocatedPorts if populated, otherwise use pre 0.12 logic
 			// COMPAT(1.0): Remove when network resources struct is removed.
 			if len(alloc.AllocatedResources.Shared.Ports) > 0 {
-				if idx.AddReservedPorts(alloc.AllocatedResources.Shared.Ports) {
+				if c, r := idx.AddReservedPorts(alloc.AllocatedResources.Shared.Ports); c {
 					collide = true
+					reason = fmt.Sprintf("collision when reserving port for alloc %s: %v", alloc.ID, r)
 				}
 			} else {
 				// Add network resources that are at the task group level
 				if len(alloc.AllocatedResources.Shared.Networks) > 0 {
 					for _, network := range alloc.AllocatedResources.Shared.Networks {
-						if idx.AddReserved(network) {
+						if c, r := idx.AddReserved(network); c {
 							collide = true
+							reason = fmt.Sprintf("collision when reserving port for network %s in alloc %s: %v", network.IP, alloc.ID, r)
 						}
 					}
 				}
 
-				for _, task := range alloc.AllocatedResources.Tasks {
-					if len(task.Networks) == 0 {
+				for task, resources := range alloc.AllocatedResources.Tasks {
+					if len(resources.Networks) == 0 {
 						continue
 					}
-					n := task.Networks[0]
-					if idx.AddReserved(n) {
+					n := resources.Networks[0]
+					if c, r := idx.AddReserved(n); c {
 						collide = true
+						reason = fmt.Sprintf("collision when reserving port for network %s in task %s of alloc %s: %v", n.IP, task, alloc.ID, r)
 					}
 				}
 			}
 		} else {
 			// COMPAT(0.11): Remove in 0.11
-			for _, task := range alloc.TaskResources {
-				if len(task.Networks) == 0 {
+			for task, resources := range alloc.TaskResources {
+				if len(resources.Networks) == 0 {
 					continue
 				}
-				n := task.Networks[0]
-				if idx.AddReserved(n) {
+				n := resources.Networks[0]
+				if c, r := idx.AddReserved(n); c {
 					collide = true
+					reason = fmt.Sprintf("(deprecated) collision when reserving port for network %s in task %s of alloc %s: %v", n.IP, task, alloc.ID, r)
```

Comment:
I like that you differentiated the error message! Hopefully users never see it or if they do they report it. In theory a user could get this in the absence of a bug if there's an allocation created by a 0.11 agent involved, but here's hoping no one is running 0.11 anything with 1.x. :crossed_fingers:

Reply:
I wasn't even sure how this path could be reached 😅 I think some of the deprecated code has already been removed.

So yeah, hopefully nobody will ever see this.